src/OnlineRetailPortal.Web/Validations/AddProductRequestValidator.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentValidation;
+using FluentValidation.Results;
+
+namespace OnlineRetailPortal.Web.Validations
+{
+    public class AddProductRequestValidator : AbstractValidator<AddProductRequest>
+    {
+        public AddProductRequestValidator()
+        {
+            RuleFor(x => x.UserId)
+            .NotEmpty()
+            .WithMessage("The User ID  cannot be blank.");
```

Comment:
What if the userId field is null?
You should also check for null
Also we need to discuss the format of how are we collecting all these messages and returning as a part if error response

Reply:
ok